src/Compilers/CSharp/Portable/Syntax/SyntaxFactory.cs

```diff
@@ -640,6 +642,492 @@ public static SyntaxToken XmlTextLiteral(SyntaxTriviaList leading, string text,
+        /// <param name="textTokens">A list of tokens used for the value of the xml text attribute.</param>
+        public static XmlTextAttributeSyntax XmlTextAttribute(XmlNameSyntax name, SyntaxKind quoteKind, SyntaxTokenList textTokens)
+        {
+            return XmlTextAttribute(
+                name,
+                Token(quoteKind),
+                textTokens,
+                Token(quoteKind))
+                .WithLeadingTrivia(Whitespace(" "));
+        }
+
+        /// <summary>
+        /// Creates the syntax representation of an xml element that spans multiple text lines.
+        /// </summary>
+        /// <param name="localName">The name of the xml element.</param>
+        /// <param name="content">A list of syntax nodes that represents the content of the xml multi line element.</param>
+        public static XmlElementSyntax XmlMultiLineElement(string localName, SyntaxList<XmlNodeSyntax> content)
+        {
+            return XmlMultiLineElement(XmlName(localName), content);
+        }
+
+        /// <summary>
+        /// Creates the syntax representation of an xml element that spans multiple text lines.
+        /// </summary>
+        /// <param name="name">The name of the xml element.</param>
+        /// <param name="content">A list of syntax nodes that represents the content of the xml multi line element.</param>
+        public static XmlElementSyntax XmlMultiLineElement(XmlNameSyntax name, SyntaxList<XmlNodeSyntax> content)
+        {
+            return XmlElement(
+                XmlElementStartTag(name),
+                content.Insert(0, XmlNewLine()).Add(XmlNewLine()),
+                XmlElementEndTag(name));
+        }
+
+        /// <summary>
+        /// Creates the syntax representation of an xml text that contains a newline token with a documentation comment 
+        /// exterior trivia at the end (continued documentation comment).
+        /// </summary>
+        public static XmlTextSyntax XmlNewLine()
+        {
+            return XmlText(XmlTextNewLine());
+        }
+
+        /// <summary>
+        /// Creates the syntax representation of an xml newline token with a documentation comment exterior trivia at 
+        /// the end (continued documentation comment).
+        /// </summary>
+        public static SyntaxToken XmlTextNewLine()
+        {
+            return XmlTextNewLine(true);
+        }
+
+        /// <summary>
+        /// Creates the syntax representation of an xml newline token for xml documentation comments.
+        /// </summary>
+        /// <param name="continueXmlDocumentationComment">
+        /// If set to true, a documentation comment exterior token will be added to the trailing trivia
+        /// of the new token.</param>
+        public static SyntaxToken XmlTextNewLine(bool continueXmlDocumentationComment)
+        {
+            SyntaxToken token = XmlTextNewLine(
+                TriviaList(),
+                Environment.NewLine,
```

Comment:
This kind of thing concerns me. It could easily introduce an implicit platform dependency. In order to test transforms or codegen that use this API you had to explicitly add Environment.NewLine to your strings instead of using simple multiline strings. Others might not be so aware. Any downstream code making assumptions about character offsets or string lengths will need to take this into account. I don't think we have precedent elsewhere in the API for using Environment.NewLine. For example, the existing EndOfLineTrivia factory methods always take an explicit string specifying their text. At a minimum I think you should follow suit here.